auth/git/credentials.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2024 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"context"
+	"time"
+
+	"github.com/fluxcd/pkg/auth"
+	"github.com/fluxcd/pkg/auth/azure"
+	"github.com/fluxcd/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// AuthOptions contains options that can be used for authentication.
+type AuthOptions struct {
+	// ProviderOptions specifies the options to configure various authentication
+	// providers.
+	ProviderOptions ProviderOptions
+
+	// Cache is a cache for storing auth configurations.
+	Cache cache.Expirable[cache.StoreObject[Credentials]]
+}
+
+// ProviderOptions contains options to configure various authentication
+// providers.
+type ProviderOptions struct {
+	AzureOpts []azure.ProviderOptFunc
+}
+
+// Credentials contains authentication data needed in order to access a Git
+// repository.
+type Credentials struct {
+	BearerToken string `json:"bearerToken,omitempty"`
+	ExpiresOn   time.Time
+}
+
+// ToSecretData returns the Credentials object in the format of the data found
+// in Kubernetes Generic Secret.
+func (c *Credentials) ToSecretData() map[string][]byte {
+	var data map[string][]byte = make(map[string][]byte)
+
+	if c.BearerToken != "" {
+		data["bearerToken"] = []byte(c.BearerToken)
+	}
+
+	return data
+}
+
+func CacheCredentials(ctx context.Context, url string, authOpts *AuthOptions, creds *Credentials) {
```

Comment:
Is this necessary? The caller provides the cache. If we want to verify the `token` before caching it, we can do that directly from the caller without relying on a function in this pkg. This is true as well if we go forward and decide that the client is solely responsible for caching.